yb-voyager/src/cp/control_plane.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cp
+
+import "github.com/google/uuid"
+
+type ControlPlane interface {
+	Init() error
+	ExportSchemaStarted(*ExportSchemaEvent)
+	ExportSchemaCompleted(*ExportSchemaEvent) // Only success is reported.
+
+	// SubmitMigrationAssessmentReport(*MigrationAssessmentReport)
+
+	SchemaAnalysisStarted(*SchemaAnalysisEvent)
+	SubmitSchemaAnalysisReport(*SchemaAnalysisReport)
```

Comment:
`SchemaAnalysisStarted` event was to signal yugabyted-ui that the analysis has started when the user runs the voyager cmd. In case of absence of `SchemaAnalysisStarted`, if due to any unknown error the analysis report was not written to the database, yugabyted-ui will not show the start event also.

Will change `SubmitSchemaAnalysisReport` to `SchemaAnalysisIterationCompleted`